### ISSUE TITLE
conan proxy dont' work/partial work

### ISSUE THREAD

**Issue by unix196** [sonatype-nexus-community/nexus-repository-conan#56]
Good day. I try explain my trouble:
```
git clone https://github.com/sonatype-nexus-community/nexus-repository-conan
docker build -t nexus-repository-conan:0.0.6 .
docker run -d -p 8081:8081 --name nexus nexus-repository-conan:0.0.6
```
Then I create conan proxy repo:
![nexus-conan-proxy](http://www.imageup.ru/img144/3340102/conan-proxy-settings.png)

Then try it using (I run command from other docker container => I use my local ip for communication with nexus instead localhost), conan-1.14.1:
```
conan remote remove conan-center
conan remote add my-proxy http://my_localip:8081/repository/test-my/ False
root@8b70608fd3c0:/# conan install arduinojson/5.13.0@conan/stable
Auto detecting your dev setup to initialize the default profile (/root/.conan/profiles/default)
Found gcc 4.9
Default settings
	os=Linux
	os_build=Linux
	arch=x86_64
	arch_build=x86_64
	compiler=gcc
	compiler.version=4.9
	compiler.libcxx=libstdc++
	build_type=Release
*** You can change them in /root/.conan/profiles/default ***
*** Or override with -s compiler='other' -s ...s***


Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=4.9
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

arduinojson/5.13.0@conan/stable: Not found in local cache, looking in remotes...
arduinojson/5.13.0@conan/stable: Trying with 'my-proxy'...
Downloading conanmanifest.txt
[==================================================] 114B/114B        
Downloading conanfile.py
[==================================================] 809B/809B        
arduinojson/5.13.0@conan/stable: Downloaded recipe revision 0
Installing package: arduinojson/5.13.0@conan/stable
Requirements
    arduinojson/5.13.0@conan/stable from 'my-proxy' - Downloaded
Packages
    arduinojson/5.13.0@conan/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download

arduinojson/5.13.0@conan/stable: Retrieving package 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 from remote 'my-proxy' 
ERROR: Binary package not found: 'arduinojson/5.13.0@conan/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9'
```
In browse repo I see that 
![browse_repo_conan_proxy_fail_install_conan1141](http://www.imageup.ru/img144/3340103/browse_repo_conan_proxy_fail_install_conan1141.png)

Then I try downgrade conan version:
```
pip install conan==1.13.3
conan install arduinojson/5.13.0@conan/stable
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=4.9
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

arduinojson/5.13.0@conan/stable: WARN: Package is corrupted, removing folder: /root/.conan/data/arduinojson/5.13.0/conan/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
Installing package: arduinojson/5.13.0@conan/stable
Requirements
    arduinojson/5.13.0@conan/stable from 'my-proxy' - Cache
Packages
    arduinojson/5.13.0@conan/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Download

arduinojson/5.13.0@conan/stable: Retrieving package 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 from remote 'my-proxy' 
Downloading conanmanifest.txt
[==================================================] 6.8KB/6.8KB      
Downloading conaninfo.txt
[==================================================] 150B/150B        
Downloading conan_package.tgz
[==================================================] 104.4KB/104.4KB  
Decompressing conan_package.tgz: 102kB [00:00, 2.73MB/s]                   
arduinojson/5.13.0@conan/stable: Package installed 5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
```
All good  
![browse_repo_conan_proxy_succes_install_conan1131](http://www.imageup.ru/img144/3340104/browse_repo_conan_proxy_succes_install_conan1131.png)

After that I try install other package with the same version of conan:
```
root@8b70608fd3c0:/# conan install zlib/1.2.11@conan/stable
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++
compiler.version=4.9
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

zlib/1.2.11@conan/stable: Not found in local cache, looking in remotes...
zlib/1.2.11@conan/stable: Trying with 'my-proxy'...
Downloading conanmanifest.txt
[==================================================] 345B/345B        
Downloading conanfile.py
[==================================================] 7.9KB/7.9KB      
Downloading conan_export.tgz
ERROR: Unable to find 'zlib/1.2.11@conan/stable' in remotes
```
In browser repo I see this - 
![browse_repo_conan_proxy_fail_install_conan1131](http://www.imageup.ru/img144/3340105/browse_repo_conan_proxy_fail_install_conan1131.png)

If I use default repository conan-center (without nexus conan proxy) - this packages install without problem. 
Is it problem conan or nexus-repositgory-conan?
With this - https://docs.conan.io/en/latest/mastering/logging.html I can debug the side between conan client and nexus-conan-proxy, but other side - between nexus-conan-proxy and remote repository https://conan.bintray.com - can't imagine how (why nexus-conan didn't download some files?) (MITM is not easy way)

**Comment by maufl** [comment]
I have the same problem/a similar problem. With the latest version of conan, the download of packages does not work.

I discovered that it will first GET the download_urls, then the conaninfo.txt but then not the conan_package.tgz but just the plain package hash.

```
{"_action": "REST_API_CALL", "duration": 0.015997886657714844, "headers": {"Authorization": "**********", "User-Agent": "Conan/1.14.3 (Python 3.7.3) python-requests/2.21.0", "X-Client-Anonymous-Id": "**********", "X-Client-Id": ""}, "method": "GET", "time": 1556087575.6687748, "url": "************/repository/hpctb-conan/v1/ping"}
{"_action": "REST_API_CALL", "duration": 0.007998466491699219, "headers": {"Authorization": "**********", "User-Agent": "Conan/1.14.3 (Python 3.7.3) python-requests/2.21.0", "X-Client-Anonymous-Id": "**********", "X-Client-Id": ""}, "method": "GET", "time": 1556087575.682775, "url": "************/repository/hpctb-conan/v1/conans/mw-sdk/2019.04-beta/fh-ipt/experimental/packages/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/download_urls"}
{"_action": "REST_API_CALL", "duration": 0.006999015808105469, "headers": {"Authorization": "**********", "User-Agent": "Conan/1.14.3 (Python 3.7.3) python-requests/2.21.0", "X-Client-Anonymous-Id": "**********"}, "method": "GET", "time": 1556087575.694775, "url": "************/repository/hpctb-conan/v1/conans/fh-ipt/mw-sdk/2019.04-beta/experimental/packages/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/conaninfo.txt"}
{"_action": "DOWNLOAD", "duration": 0.01200103759765625, "time": 1556087575.6997771, "url": "************/repository/hpctb-conan/v1/conans/fh-ipt/mw-sdk/2019.04-beta/experimental/packages/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7/conaninfo.txt"}
{"_action": "REST_API_CALL", "duration": 0.008002996444702148, "headers": {"Authorization": "**********", "User-Agent": "Conan/1.14.3 (Python 3.7.3) python-requests/2.21.0", "X-Client-Anonymous-Id": "**********", "X-Client-Id": ""}, "method": "GET", "time": 1556087575.7377355, "url": "************/repository/hpctb-conan/v1/conans/mw-sdk/2019.04-beta/fh-ipt/experimental/packages/6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7"}
{"_action": "EXCEPTION", "class": "PackageNotFoundException", "message": "Binary package not found: 'mw-sdk/2019.04-beta@fh-ipt/experimental:6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7'", "time": 1556087575.7457314}
```

**Comment by unix196** [comment]
I wrote other issue - https://github.com/conan-io/conan/issues/5027 in conan github repository -there I can try mixing proxy repository and conan-center for download packages - also don't work.

**Comment by memsharded** [comment]
Conan has recently migrated to use "revisions" (automatic internal version of packages that have the same version), using a new v2 API that support those revisions. All the old APIs still work (v1), but it is possible that it is affecting somehow.

**Comment by doddi** [comment]
If you are feeling brave I would be interested to see if this branch fixes your issue https://github.com/sonatype-nexus-community/nexus-repository-conan/tree/refactor-v1-api

**Comment by igorudovika** [comment]
Main part of refactor-v1-api is in master now. In same time we are not facing now with mentioned issue in current released version(1.0.2).
